### ISSUE TITLE
Error while restoring a webapp

### ISSUE THREAD

**Issue by kishorebasisth** [Azure/azure-cli#4588]
### Description
Since a week now, I am getting an error while trying to restore a webapp from the backup saved in the Storage account.

Following is the command that I used to do so:

```
az webapp config backup restore --backup-name $backupname --container-url $ContainerURL --resource-group group-name --webapp-name some-web-app --overwrite --ignore-hostname-conflict
```
The error is:

```
Unable to determine app service plan for restored site. Specify an app service plan.
```
There's no option to pass the app service plan as a command line argument for the restore command.

The detailed log snippet in the debug mode is as follow:

```
urllib3.connectionpool : Starting new HTTPS connection (1): management.azure.com
urllib3.connectionpool : https://management.azure.com:443 "POST /subscriptions/xxxxxx-xxxxx-xxxxx-9190-xxxxxxxxx/resourceGroups/groups/providers/Microsoft.Web/sites/somerestore/backups/0/restore?api-version=2016-08-01 HTTP/1.1" 400 562
msrest.http_logger : Request URL: 'https://management.azure.com/subscriptions//xxxxxx-xxxxx-xxxxx-9190-xxxxxxxxx/resourceGroups/groups/providers/Microsoft.Web/sites/somerestore/backups/0/restore?api-version=2016-08-01'
msrest.http_logger : Request method: 'POST'
msrest.http_logger : Request headers:
msrest.http_logger :     'Connection': 'keep-alive'
msrest.http_logger :     'Accept-Encoding': 'gzip, deflate'
msrest.http_logger :     'Accept': 'application/json'
msrest.http_logger :     'User-Agent': 'python/2.7.12 (Linux-4.10.0-35-generic-x86_64-with-Ubuntu-16.04-xenial) requests/2.18.1 msrest/0.4.11 msrest_azure/0.4.9 websitemanagementclient/0.32.0 Azure-SDK-For-Python AZURECLI/2.0.10'
msrest.http_logger :     'Authorization': '*****'
msrest.http_logger :     'x-ms-client-request-id': 'fd4f62f4-a50e-11e7-9c25-080027f09b28'
msrest.http_logger :     'CommandName': 'webapp config backup restore'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'Content-Length': '329'
msrest.http_logger :     'accept-language': 'en-US'
msrest.http_logger : Request body:
msrest.http_logger : {"properties": {"ignoreConflictingHostNames": true, "blobName": "zzxxxccccc.zip", "storageAccountUrl": "https://account-URL", "operationType": "Default", "overwrite": true}, "location": "West Europe"}
msrest.http_logger : Response status: 400
msrest.http_logger : Response headers:
msrest.http_logger :     'Cache-Control': 'no-cache'
msrest.http_logger :     'Pragma': 'no-cache'
msrest.http_logger :     'Content-Length': '562'
msrest.http_logger :     'Content-Type': 'application/json; charset=utf-8'
msrest.http_logger :     'Expires': '-1'
msrest.http_logger :     'ETag': '"1D33919FAE4B055"'
msrest.http_logger :     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
msrest.http_logger :     'Server': 'Microsoft-IIS/8.0'
msrest.http_logger :     'X-AspNet-Version': '4.0.30319'
msrest.http_logger :     'X-Powered-By': 'ASP.NET'
msrest.http_logger :     'x-ms-ratelimit-remaining-subscription-writes': '1199'
msrest.http_logger :     'x-ms-request-id': 'd3a13a83-1170-44dd-8ce5-2ce702f72074'
msrest.http_logger :     'x-ms-correlation-request-id': 'd3a13a83-1170-44dd-8ce5-2ce702f72074'
msrest.http_logger :     'x-ms-routing-request-id': 'WESTEUROPE:20170929T120923Z:d3a13a83-1170-44dd-8ce5-2ce702f72074'
msrest.http_logger :     'Date': 'Fri, 29 Sep 2017 12:09:23 GMT'
msrest.http_logger : Response content:
msrest.http_logger : {"Code":"BadRequest","Message":"Unable to determine app service plan for restored site. Specify an app service plan.","Target":null,"Details":[{"Message":"Unable to determine app service plan for restored site. Specify an app service plan."},{"Code":"BadRequest"},{"ErrorEntity":{"ExtendedCode":"54110","MessageTemplate":"Unable to determine app service plan for restored site. Specify an app service plan.","Parameters":[],"Code":"BadRequest","Message":"Unable to determine app service plan for restored site. Specify an app service plan."}}],"Innererror":null}
msrest.exceptions : Operation failed with status: 'Bad Request'. Details: 400 Client Error: Bad Request for url: https://management.azure.com/subscriptions//xxxxxx-xxxxx-xxxxx-9190-xxxxxxxxx/resourceGroups/groups/providers/Microsoft.Web/sites/somerestore/backups/0/restore?api-version=2016-08-01
Unable to determine app service plan for restored site. Specify an app service plan.

```
---

### Environment summary

**Install Method:** How did you install the CLI? (e.g. pip, interactive script, apt-get, Docker, MSI, nightly)  
Answer here:

`curl -L https://aka.ms/InstallAzureCli | bash`

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
Answer here:

```
azure-cli (2.0.9)

acr (2.0.7)
acs (2.0.9)
appservice (0.1.9)
batch (3.0.2)
billing (0.1.2)
cdn (0.0.5)
cloud (2.0.5)
cognitiveservices (0.1.5)
command-modules-nspkg (2.0.0)
component (2.0.6)
configure (2.0.9)
consumption (0.1.2)
core (2.0.10)
cosmosdb (0.1.9)
dla (0.0.9)
dls (0.0.9)
feedback (2.0.5)
find (0.2.5)
interactive (0.3.5)
iot (0.1.8)
keyvault (2.0.7)
lab (0.0.7)
monitor (0.0.7)
network (2.0.9)
nspkg (3.0.0)
profile (2.0.7)
rdbms (0.0.4)
redis (0.2.6)
resource (2.0.9)
role (2.0.7)
sf (1.0.4)
sql (2.0.6)
storage (2.0.9)
vm (2.0.9)

Python (Linux) 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]

```

**OS Version:** What OS and version are you using?  
Answer here:

`39~16.04.1-Ubuntu SMP Wed Sep 13 09:02:42 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
Answer here:

`zsh`

**Comment by tjprescott** [comment]
@yugangw-msft can you take a look?

**Comment by yugangw-msft** [comment]
It is broken at the service end.
I am following up with service end folks

**Comment by derekbekoe** [comment]
@yugangw-msft I see `https://management.azure.com/subscriptions//xxxxxx-xxxxx-xxxxx-9190-xxxxxxxxx/resourceGroups/groups/providers/Microsoft.Web/sites/somerestore/backups/0/restore?api-version=2016-08-01`. Note the `//`.

Could that be the cause of the issue?

**Comment by yugangw-msft** [comment]
@derekbekoe, that is just a bad error. The real cause was a breaking change at back-end.
The fix was submitted: https://github.com/Azure/azure-cli/pull/4604/files#diff-35983163c0f2f417dda0c571d4a4538dR827

**Comment by yugangw-msft** [comment]
@kishorebasisth, the fix was merged through #4604 and will be released at around Oct 23 . You can try our nightly or docker image before the release

**Comment by yugangw-msft** [comment]
//cc @nking92 @LukaszStem
